### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by ccheraa**
Packages updated:
- filesize: 6.3.0 => 7.0.0
- typescript: 4.2.4 => 4.3.5
- mocha: 8.4.0 => 9.0.2
- sinon: 10.0.0 => 11.1.1
- ts-node: 9.1.1 => 10.1.0

### CODE REVIEW

File: cypress/support/commands.js [hunk 1]
```diff
@@ -16,16 +16,26 @@
...
     let obj = {}
     obj[conn_val] = nested
 
-    localStorage.setItem('connections',	JSON.stringify(obj))
+    localStorage.setItem('connections', JSON.stringify(obj))
 
- })
+})
+Cypress.Commands.overwrite('visit', (orig, url, options) => {
+    // this is a fix to include the process variable when using the Electron browser
```
Comment (duncdrum) [inline cypress/support/commands.js:30]: can you be more specific here, links to the relevant issues and commits would be helpful too, as we need to maintain and understand this down the road

File: cypress/plugins/index.js [hunk 2]
```diff
@@ -18,4 +18,45 @@
 module.exports = (on, config) => {
   // `on` is used to hook into various events Cypress emits
   // `config` is the resolved Cypress config
+  config.process = {
```
Comment (duncdrum) [inline cypress/plugins/index.js:21]: do we really need to redefine all process variables like this or can we do this in a more targeted fashion? Also we need some form of documentation of why we started doing this and what its supposed to achieve. Again links to open issues would be a great help here.

**Review by duncdrum — COMMENTED** [review]
we still have 0 tests passing, where we had three passing before we merged dependency updates. My main concern is with the maintainability of these changes. 

We seem to be tacking a shotgun approach by modifying `cy.visit` and adding a bunch of variables, when all we apparently need is a single process variable to be added when the browser app is tested in electron. 

the changes need more documentation and links to either webpack/theia/cypress/fusion-studio issues so that we can understand why these changes where introduced in the first place and what they are supposed to be doing.

File: cypress/plugins/index.js (hunk 2 above)
Comment (ccheraa) [reply cypress/plugins/index.js:21]: It looks like my rebase wasn't perfect after all, I'll fix that.

File: cypress/support/commands.js (hunk 1 above)
Comment (ccheraa) [reply cypress/support/commands.js:30]: I added a link to the issue on Theia's repo

**Comment by ccheraa** [comment]
I have run all the tests in headed and headless with all the browser, and they all passed
I'm using FusionDB nightly build
```
System:
 - Ubuntu 20.04
Api:
 - Version: 1.1.1
Server:
 - Product: FusionDB
 - Version: 1.0.0-SNAPSHOT
 - Revision: 55c199eaf4742bb63d3e00c84c22bbc837789eb0
 - Build: 20210323185156
```

Check PR #458

**Pull request merged**